src/KahanSummation.jl

```diff
@@ -113,4 +114,27 @@ function sum_kbn(A)
     s - c
 end
 
+function sum_kbn(f::Function, A)
```

Comment:
This needn't be a function; for example, one could also pass a type.

Reply:
I know -- I do not know how to annotate Function or TypeConstructor.  When I had it as `sum_kbn(f, A)` things got murky with sum_kbn(identity, A) and there was no clean dispatch resolution. What is the correct way to approach this while following the guide that sum_kbn(A) should be defined in terms of sum_kbn(identity, A)?

Reply:
I think the issue there is that the definition with `identity` was backwards. It should be `sum_kbn(A) = sub_kbn(identity, A)`, with `sum_kbn(f, A)` as the only long-form definition of the function. That should remove any ambiguities.

Reply:
does the application of identity disappear always?

Reply:
e.g. map(identity, [....])

Reply:
Yes, the function call should get optimized out